lstchain/scripts/lstchain_create_run_summary.py

```diff
@@ -0,0 +1,309 @@
+    type=Path,
+    help="Directory in which Run Summary file is written",
+    default="./",
+)
+
+dtypes = {
+    "ucts_timestamp": np.int64,
+    "run_start": np.int64,
+    "dragon_reference_time": np.int64,
+    "dragon_reference_module_id": np.int16,
+    "dragon_reference_module_index": np.int16,
+    "dragon_reference_counter": np.uint64,
+    "dragon_reference_source": str,
+}
+
+
+def get_list_of_files(r0_path):
+    """
+    Get the list of R0 files from a given date.
+
+    Parameters
+    ----------
+    r0_path : pathlib.Path
+        Path to the R0 files
+
+    Returns
+    -------
+    list_of_files: pathlib.Path.glob
+        List of files
+    """
+    # FIXME: use regular expressions from lstchain.paths.R0_RE
+    list_of_files = r0_path.glob("LST*.fits.fz")
+    return list_of_files
+
+
+def get_list_of_runs(list_of_files):
+    """
+    Get the sorted list of run objects from R0 filenames.
+
+    Parameters
+    ----------
+    list_of_files : pathlib.Path.glob
+        List of files
+    Returns
+    -------
+    list_of_run_objects
+    """
+    return sorted(parse_r0_filename(file) for file in list_of_files)
+
+
+def get_runs_and_subruns(list_of_run_objects, stream=1):
+    """
+    Get the list of run numbers and the number of sequential files (subruns)
+    of each run.
+
+    Parameters
+    ----------
+    list_of_run_objects
+    stream: int, optional
+        Number of the stream to obtain the number of sequential files (default is 1).
+
+    Returns
+    -------
+    (run, number_of_files) : tuple
+        Run numbers and corresponding subrun of each run.
+    """
+    list_filtered_stream = filter(lambda x: x.stream == stream, list_of_run_objects)
+
+    run, number_of_files = np.unique(
+        list(map(lambda x: x.run, list_filtered_stream)), return_counts=True
+    )
+
+    return run, number_of_files
+
+
+def type_of_run(date_path, run_number, counters, n_events=500):
+    """
+    Get empirically the type of run based on the percentage of
+    pedestals/mono trigger types from the first n_events:
+    DRS4 pedestal run: 100% mono events (trigger_type == 1)
+    cosmic data run: <10% pedestal events (trigger_type == 32)
+    pedestal-calibration run: ~50% mono, ~50% pedestal events
+
+    Parameters
+    ----------
+    date_path : pathlib.Path
+        Path to the R0 files
+    run_number : int
+        Run id
+    counters : dict
+        Dict containing the reference counters and timestamps
+    n_events : int
+        Number of events used to infer the type of the run
+
+    Returns
+    -------
+    run_type: str
+        Type of run (DRS4, CALI, DATA, CONF)
+    """
+
+    list_of_files = sorted(date_path.glob(f"LST-1.1.Run{run_number:05d}.*.fits.fz"))
+    filename = list_of_files[0]
+
+    config = Config()
+    config.EventTimeCalculator.dragon_reference_time = int(counters["dragon_reference_time"])
+    config.EventTimeCalculator.dragon_reference_counter = int(counters["dragon_reference_counter"])
```

Comment:
Also set `dragon_module_id` here.

Reply:
BTW, why do we need both?
```
# - {name: dragon_reference_module_id, datatype: int16}
# - {name: dragon_reference_module_index, datatype: int16}
```

Reply:
For the application we only need the index and only that is used. For a human looking at the file I thought it might be good to give both and as input from the commandline I also though that given id rather then index is better.

Reply:
It would be just adding this line?

```python
config.EventTimeCalculator.dragon_module_id = int(counters["dragon_reference_module_id"])
```

I tried adding that and 

```python
config.EventTimeCalculator.dragon_module_id = int(counters["dragon_reference_module_index"])
```

but I get this error with the test file `test_data/real/R0/20200218/LST-1.1.Run02006.0004.fits.fz`:

```python
Files test_data/real/R0/20200218/LST-1.*.Run02006.0000*.fits.fz have error: 
Traceback (most recent call last):
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/bin/lstchain_create_run_summary", line 33, in <module>
    sys.exit(load_entry_point('lstchain', 'console_scripts', 'lstchain_create_run_summary')())
  File "/home/dmorcuende/software/lst_software/cta-lstchain/cta-lstchain/lstchain/scripts/lstchain_create_run_summary.py", line 291, in main
    for run, counters in zip(run_numbers, reference_counters)
  File "/home/dmorcuende/software/lst_software/cta-lstchain/cta-lstchain/lstchain/scripts/lstchain_create_run_summary.py", line 291, in <listcomp>
    for run, counters in zip(run_numbers, reference_counters)
  File "/home/dmorcuende/software/lst_software/cta-lstchain/cta-lstchain/lstchain/scripts/lstchain_create_run_summary.py", line 165, in type_of_run
    event_type_counts = Counter(event.trigger.event_type for event in source)
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/collections/__init__.py", line 568, in __init__
    self.update(*args, **kwds)
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/collections/__init__.py", line 655, in update
    _count_elements(self, iterable)
  File "/home/dmorcuende/software/lst_software/cta-lstchain/cta-lstchain/lstchain/scripts/lstchain_create_run_summary.py", line 165, in <genexpr>
    event_type_counts = Counter(event.trigger.event_type for event in source)
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/ctapipe/io/eventsource.py", line 274, in __iter__
    for event in self._generator():
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/ctapipe_io_lst/__init__.py", line 334, in _generator
    self.fill_trigger_info(array_event)
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/ctapipe_io_lst/__init__.py", line 499, in fill_trigger_info
    trigger.time = self.time_calculator(tel_id, array_event)
  File "/home/dmorcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/ctapipe_io_lst/event_time.py", line 240, in __call__
    module_index = np.where(lst.svc.module_ids == self.dragon_module_id.tel[tel_id])[0][0]
IndexError: index 0 is out of bounds for axis 0 with size 0
```

It simply should not go through the function since it does not fulfill the glob pattern `pattern = date_path / f"LST-1.*.Run{run_number:05d}.0000*.fits.fz"`

Reply:
> config.EventTimeCalculator.dragon_module_id = int(counters["dragon_reference_module_id"])

Yes, this should be correct.

> config.EventTimeCalculator.dragon_module_id = int(counters["dragon_reference_module_index"])

No, this is wrong, index is index and module is module.

> IndexError: index 0 is out of bounds for axis 0 with size 0

I think modules start with 1, right? so module_id == 0 leads to an empty return value from `np.where`.

Reply:
No, module id starts at 0. That is not it.

Reply:
I think that the raiseError inside the try is not going to be raised, right?

```python
    try:
        f = MultiFiles(glob(str(pattern)))
        first_event = next(f)

        if first_event.event_id != 1:
            raise ValueError("Must be used on first file streams (subrun)")
```

At least it does not with file `test_data/real/R0/20200218/LST-1.1.Run02006.0004.fits.fz` whose first event should not be 1. It just print out the line `Files test_data/real/R0/20200218/LST-1.*.Run02006.0000*.fits.fz have error: ` Hence traceback is not related to that

Reply:
subruns are only 4 zeros, right? So you're glob patter will ony match the `0000`

Reply:
```
2005,1,DATA,1582055055150584600,1582055042000000000,1582055055150584600,90,0,5150584300,ucts
2008,2,DATA,1582059789516171575,1582059778000000000,1582059789516171575,90,0,2516171300,ucts
```

Reply:
strange.

Reply:
commenting out the line of `dragon_module_id` I'm able to obtain this summary:
```
2005,1,DATA,1582055055150584600,1582055042000000000,1582055055150584600,90,0,5150584300,ucts
2006,1,DATA,-1,-1,-1,-1,-1,18446744073709551615,None
2008,2,DATA,1582059789516171575,1582059778000000000,1582059789516171575,90,0,2516171300,ucts
```

Reply:
The exception IndexError when dragon module id is set to -1 is captured now.